include/envoy/event/file_event.h

```diff
@@ -18,15 +18,28 @@ struct FileReadyType {
   static const uint32_t Closed = 0x4;
 };
 
-enum class FileTriggerType { Level, Edge };
+#define FORCE_LEVEL_EVENTS 0
+enum class FileTriggerType { Level, Edge, EmulatedEdge };
```

Comment:
Can we get rid of Level now? Windows is the only platform its needed and we fully intend to use the "optimized" flavor anyhow. Or are you using it for testing/comparison?

Reply:
Level events are used by listeners and DNS ( on all platforms), so we cant really remove it.

Reply:
Ah I did not know that. Have you played with trying to move listeners and DNS to EmulatedEdge? Just curious. This is something that can certainly be done later if there is interest

Reply:
Can you add some comments now above each trigger type with what they do and what they are used for? I think now it's less clear what all of these do and are for and it would be good to have a comment trail.

Reply:
I added a block of comments for `EmulatedEdge` and some comments with the current use cases of the other event types